src/core/linker.cpp

```diff
@@ -335,10 +335,18 @@ void Linker::InitTlsForThread(bool is_primary) {
             &addr_out, tls_aligned, 3, 0, "SceKernelPrimaryTcbTls");
         ASSERT_MSG(ret == 0, "Unable to allocate TLS+TCB for the primary thread");
     } else {
-        if (heap_api_func) {
-            addr_out = heap_api_func(total_tls_size);
-        } else {
+        if (heap_api) {
+            LOG_ERROR(Core_Linker, "TLS user malloc called, using std::malloc");
+            // TODO: ref rtld_tls_block_malloc, tls malloc replacement
+            // addr_out = heap_api->heap_malloc(total_tls_size);
             addr_out = std::malloc(total_tls_size);
```

Comment:
Can this be limited to windows only? iirc linux has no such breakage and it's probably safer to use this. malloc on linux might return high addresses